OurTimeForTech/shiftwork2
# Building Out the rest of Shift functionality 😅

## What type of PR is this? (check all applicable)



## Description of what PR does
This PR does the following:
- implements the `flatpickr` [tool](https://flatpickr.js.org/) to enable organizations to choose the start and end date-and-time for their shifts using a UX-friendly calendar picker. 
- ensures that shift times are displayed in user's chosen timezone, but stored in the database in UTC
- ensures that organizations cannot set a shift end-date before the shift start-date
- ensures that organizations create shifts of at least 1 hour
- ensures that a worker cannot drop a shift once the start-date-time is within 24 hours
- improves several related aspects of the application (eg link to worker assigned to shift)
- redirects organizations who hit `/shifts/` to their org#show page (which lists their own shifts). Keep the `/shifts` endpoint strictly for workers.
- refactors some areas (eg removes `@worker_bio` and just calls `@worker.bio`)
- ~Add number of completed shifts to worker's show page (as seen by organizations)~ (decided this was superfluous)
- fades alerts after 5 seconds
- other things I cannot think of now 😆  


## QA Instructions, Screenshots, Recordings
- Log in to app as both organization and worker (in seperate tabs) and play around with creating shifts.
- Create some shifts that start within 24 hours; next, take these shifts as a worker.
- Then, try to drop them - you shouldn't be able to.
- As an org, navigate to `/shifts`. You should be redirected to your `/organizations/:id` page.
- As an org, try to create shifts that are either less than 1 hour, or where the end occurs before the start. You should get the appropriate errors, and see that the create or edit form is re-rendered.


## Added tests?

- [X] yes, for the user parts; tests for shifts will be done by another team member.


## Added to documentation?
